sklearn/model_selection/tests/test_validation.py

```diff
@@ -915,54 +916,157 @@ def test_cross_val_predict_sparse_prediction():
 
 
 def test_cross_val_predict_with_method():
-    check_cross_val_predict_with_method(LogisticRegression())
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    X, y = shuffle(X, y, random_state=0)
+    for method in ['decision_function', 'predict_proba', 'predict_log_proba']:
+        est = LogisticRegression()
+        out = check_cross_val_predict_with_method(est, X, y, method)
+        assert_array_equal(out.shape, (len(X), len(set(y))))
 
 
 def test_gridsearchcv_cross_val_predict_with_method():
+    iris = load_iris()
+    X, y = iris.data, iris.target
+    X, y = shuffle(X, y, random_state=0)
     est = GridSearchCV(LogisticRegression(random_state=42),
                        {'C': [0.1, 1]},
                        cv=2)
-    check_cross_val_predict_with_method(est)
+    for method in ['decision_function', 'predict_proba', 'predict_log_proba']:
+        out = check_cross_val_predict_with_method(est, X, y, method)
+        assert_array_equal(out.shape, (len(X), len(set(y))))
+
+
+def test_cross_val_predict_with_method_multilabel_ovr():
+    # OVR does multilabel predictions, but only arrays of
+    # binary indicator columns. The output of predict_proba
+    # is a 2D array with shape (n_samples, n_classes).
+    n_samp = 100
+    n_classes = 4
+    X, y = make_multilabel_classification(n_samples=n_samp, n_labels=3,
+                                          n_classes=n_classes, n_features=5,
+                                          random_state=42)
+    est = OneVsRestClassifier(LogisticRegression(random_state=0))
+    for method in ['predict_proba', 'decision_function']:
+        out = check_cross_val_predict_with_method(est, X, y, method=method)
+        assert_array_equal(out.shape, (n_samp, n_classes))
+
+
+def test_cross_val_predict_with_method_multilabel_rf():
+    # The RandomForest allows anything for the contents of the labels.
+    # Output of predict_proba is a list of outputs of predict_proba
+    # for each individual label.
+    n_classes = 4
+    X, y = make_multilabel_classification(n_samples=100, n_labels=3,
+                                          n_classes=n_classes, n_features=5,
+                                          random_state=42)
+    y[:, 0] += y[:, 1]  # Put three classes in the first column
+    est = RandomForestClassifier(n_estimators=5, random_state=0)
+    out = check_cross_val_predict_with_method(est, X, y, method='predict_proba')
+    assert_array_equal(out[0].shape, (len(X), 3))
+    for i_output in range(1, n_classes):
+        assert_array_equal(out[i_output].shape, (len(X), 2))
+
+
+def test_cross_val_predict_with_method_rare_class():
+    # Test a multiclass problem where one class will be missing from
+    # one of the CV training sets.
+    rng = np.random.RandomState(0)
+    X = rng.normal(0, 1, size=(10, 10))
+    y = np.array([0, 1, 0, 1, 0, 1, 0, 1, 0, 2])
+    est = LogisticRegression()
+    for method in ['predict_proba', 'predict_log_proba']:
```

Comment:
I'd left it off because the test code doesn't handle the decision_function case when there's three classes in the full data but only 2 in one of the folds. I think that case is covered by `test_cross_val_predict_class_subset`. In this test, I modified `y` to have 4 classes instead of 3 and added "decision_function" to the list.